Caltech-IPAC/firefly
# Firefly-527: Web API Infrastructure

### Firefly-527: Web API Infrastructure

  - Infrastructure done
  - Help page done
  - API is created for each application, not there by default
  - 4 searches,
      - image
      - HiPs
      - table
      - tap panel
  - Auto execute of tap panel
  - Updated showDropDown so it now fully supports initArgs
  - Now uses `api=<command>`

_ticket:_ https://jira.ipac.caltech.edu/browse/FIREFLY-527
_ife PR:_ https://github.com/IPAC-SW/irsa-ife/pull/112

### Testing
_firefly:_ https://fireflydev.ipac.caltech.edu/firefly-527-url-api/firefly/
_irsaviewer:_ https://irsakudevlb1.ipac.caltech.edu/firefly-527-url-api/irsaviewer/
_sofia:_ https://irsakudevlb1.ipac.caltech.edu/firefly-527-url-api/applications/sofia/

adding   `?api`  will get you the help page

_firefly:_ https://fireflydev.ipac.caltech.edu/firefly-527-url-api/firefly/?api
_irsaviewer:_ https://irsakudevlb1.ipac.caltech.edu/firefly-527-url-api/irsaviewer/?api
_sofia:_ https://irsakudevlb1.ipac.caltech.edu/firefly-527-url-api/applications/sofia/?api


- try the URLs in the document
- copy a URL and add parameters to test errors
- copy a url change the command to test errors
- try other parameters from the document
- notice IRSAViewer does not have tap
- includes a starting implementation for the sofia web API
